app/services/teachers/school_transfers/history.rb

```diff
@@ -0,0 +1,72 @@
+module Teachers::SchoolTransfers
+  class History
+    ALLOWED_TEACHER_TYPES = %i[ect mentor].freeze
+
+    Transfer = Data.define(:leaving_training_period, :joining_training_period) do
+      def type
+        return :unknown unless joining_training_period
+        return :new_provider if leaving_training_period.lead_provider !=
+          joining_training_period.lead_provider
+
+        :new_school
+      end
+
+      def status
+        return :complete unless leaving_training_period.finished_on.future? ||
+          joining_training_period&.started_on&.future?
+
+        :incomplete
+      end
+
+      def teacher_type = leaving_training_period.for_ect? ? :ect : :mentor
+    end
```

Comment:
I put this in here for now, but we should probably extract it to it's own class and add some tests.